platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java

```diff
@@ -517,16 +519,24 @@ public class SwirldsPlatform implements Platform {
         signedStateFileManagerWiring.solderStatusManager(platformStatusManager);
         signedStateFileManagerWiring.solderAppCommunication(appCommunicationComponent);
 
+        final LatestCompleteStateNexus latestCompleteState =
+                new LatestCompleteStateNexus(stateConfig, platformContext.getMetrics());
         final SavedStateController savedStateController =
                 new SavedStateController(stateConfig, signedStateFileManagerWiring.saveStateToDisk()::offer);
-
+        final NewLatestCompleteStateConsumer newLatestCompleteStateConsumer = ss -> {
+            // the app comm component will reserve the state, this should be done by the wiring in the future
+            appCommunicationComponent.newLatestCompleteStateEvent(ss);
+            // the nexus expects a state to be reserved for it
+            // in the future, all of these reservations will be done by the wiring
+            latestCompleteState.setState(ss.reserve("setting latest complete state"));
+        };
```

Comment:
IMO, this is the type of data passing that is best done inside the wiring framework and not via lambdas. Ok if this is a temporary solution you plan on fixing later, but want to make sure this is not our permanent solution.

Reply:
this is the plan, but we have to get there step by step